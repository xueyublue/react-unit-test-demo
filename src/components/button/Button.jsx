import React from "react";

export default function Button({ label }) {
  return <button data-testid="button">{label}</button>;
}
