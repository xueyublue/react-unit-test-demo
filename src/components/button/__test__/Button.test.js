import { render, screen, cleanup } from "@testing-library/react";
import Button from "../Button";
import renderer from "react-test-renderer";

beforeEach(cleanup);

test("render success", () => {
  render(<Button label="SAVE"></Button>);
  const button = screen.getByTestId("button");
  expect(button).toHaveTextContent("SAVE");
});

test("render snapshot", () => {
  const tree = renderer.create(<Button label="SAVE" />).toJSON();
  expect(tree).toMatchSnapshot();
});
