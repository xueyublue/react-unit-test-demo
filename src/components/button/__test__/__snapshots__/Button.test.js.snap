// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<button
  data-testid="button"
>
  SAVE
</button>
`;
